### PR TITLE
Replace API endpoint and speech synthesizer

### PR DESCRIPTION
**Pull request by johnheusinger**
# PR Summary

I noticed that the 'Fun with Cat Facts' advice snippet referenced a REST API that is no longer available. I found a replacement API and made some adjustments so that the result matches the original intention.

I also noticed that the example makes use of System.Speech which works on Windows PowerShell but not PowerShell Core. I switched to the SAPI.SpVoice COM object which is also compatible with PS Core.

## Context

Relates to #260 

## Changes

- Replaced unavailable API endpoint with a working one
- Replaced System.Speech with SAPI.SpVoice for PS Core compatibility
- Spelling and formatting corrections

## Checklist

- [X] Pull Request has a meaningful title.
- [X] Summarised changes.
- [X] Pull Request is ready to merge & is not WIP.
- [ ] Added tests / only testable interactively.
  - Make sure you add a new test if old tests do not effectively test the code changed.
- [ ] Added documentation / opened issue to track adding documentation at a later date.

### CODE REVIEW

File: PSKoans/Data/Advice/Fun/FunWithCats.Advice.json [hunk 1]
```diff
@@ -5,20 +5,17 @@
...
-    "\tAdd-Type -AssemblyName System.Speech",
-    "\t$SpeechSynth = New-Object System.Speech.Synthesis.SpeechSynthesizer",
-    "\t$URI = 'http://www.catfact.info/api/v1/facts.json?per_page=200'",
-    "\t$Random = Get-Random -Maximum 200",
-    "\t$CatFact = (ConvertFrom-Json (Invoke-WebRequest -Uri $URI)).Facts[$Random]",
-    "\t$SpeechSynth.Speak('did you know?')",
-    "\t$SpeechSynth.Speak($CatFact.details)",
+    "\t$SpeechSynth = New-Object -ComObject SAPI.SpVoice",
+    "\t$URI = 'https://catfact.ninja/fact'",
+    "\t$CatFact = Invoke-WebRequest -Uri $URI | ConvertFrom-Json | Select-Object -ExpandProperty fact",
```
Comment (vexx32) [inline PSKoans/Data/Advice/Fun/FunWithCats.Advice.json:17]:
```suggestion
    "\t$CatFact = Invoke-RestMethod -Uri $URI | Select-Object -ExpandProperty fact",
```

IRM does a direct JSON conversion for this purpose 🙂

**Review by vexx32 — CHANGES_REQUESTED** [review]
Looks good to me apart from one or two small things to adjust.

Thanks for sorting this out! 💖 😊

File: PSKoans/Data/Advice/Fun/FunWithCats.Advice.json [hunk 2]
```diff
@@ -5,20 +5,17 @@
...
+    "\t$SpeechSynth.Speak(\"Did you know? $Catfact\") | Out-Null",
```
[the remark this answers is not in the document]
Comment (johnheusinger) [reply PSKoans/Data/Advice/Fun/FunWithCats.Advice.json]: Done 🙂

File: PSKoans/Data/Advice/Fun/FunWithCats.Advice.json (hunk 1 above)
Comment (johnheusinger) [reply PSKoans/Data/Advice/Fun/FunWithCats.Advice.json]: And done. Thanks 🙂

**Review by vexx32 — APPROVED** [review]

**Pull request merged**